### PR TITLE
Set exposeHeadRoutes: true by default

### PR DESCRIPTION
**Pull request by mcollina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/route.js [hunk 1]
```diff
@@ -227,9 +227,7 @@ function buildRouting (options) {
         }
       }
 
-      const config = opts.config || {}
-      config.url = url
-      config.method = opts.method
+      const config = Object.assign({ url, method: opts.method }, opts.config)
```
Comment (mcollina) [inline lib/route.js:230]: This is a bad bug that's present on master as well.

**Comment by mcollina** [comment]
I took the lazy route and disabled the HEAD routes in certain tests. Updating them would have made more complex to read from my point of view.

File: fastify.js [hunk 2]
```diff
@@ -130,10 +129,12 @@ function fastify (options) {
   options.disableRequestLogging = disableRequestLogging
   options.ajv = ajvOptions
   options.clientErrorHandler = options.clientErrorHandler || defaultClientErrorHandler
-  options.exposeHeadRoutes = exposeHeadRoutes
 
   const initialConfig = getSecuredInitialConfig(options)
 
+  // TODO remove this, it should not be needed
```
Comment (L2jLiga) [inline fastify.js:135]: Does todo means that PR is not yet ready?

Comment (mcollina) [reply fastify.js:135]: no, I'll remove the todo.

**Review by fox1t — APPROVED** [review]

File: lib/route.js (hunk 1 above)
Comment (fox1t) [reply lib/route.js:230]: Nice finding!

**Review by zekth — APPROVED** [review]

**Review by L2jLiga — APPROVED** [review]
LGTM

File: lib/route.js (hunk 1 above)
Comment (mcollina) [reply lib/route.js:230]: Fixed in master with #2827

**Review by jsumners — APPROVED** [review]
LGTM

**Review by salmanm — APPROVED** [review]

**Comment by mcollina** [comment]
PTAL, I've rebased.

**Review by salmanm — APPROVED** [review]
lgtm.
Just for my curiosity and since there isn't any issue linked, would like to know what's the motivation behind it?

**Comment by zekth** [comment]
> lgtm.
> Just for my curiosity and since there isn't any issue linked, would like to know what's the motivation behind it?

Following this one: https://github.com/fastify/fastify/pull/2700

**Review by jsumners — APPROVED** [review]

**Pull request merged**